plugins/generator-1.21.1/neoforge-1.21.1/procedures/gui_set_items.java.ftl

```diff
@@ -1,7 +1,8 @@
 <#include "mcitems.ftl">
-if(${input$entity} instanceof Player _player && _player.containerMenu instanceof ${JavaModName}Menus.MenuAccessor _menu) {
-	ItemStack _setstack = ${mappedMCItemToItemStackCode(input$item, 1)}.copy();
-	_setstack.setCount(${opt.toInt(input$amount)});
-	_menu.getSlots().get(${opt.toInt(input$slotid)}).set(_setstack);
+<@head>if(${input$entity} instanceof Player _player && _player.containerMenu instanceof ${JavaModName}Menus.MenuAccessor _menu) {</@head>

```

Comment:
Same here, the head is dependent on item, amount, and slotid. It is highly unlikely user will call this procedure block multiple times for the same item, amount, and slotid. Optimize this to the outer if only as this way much broader range of blocks that have same if and end with broadcastChanges will be mergable.

If your head is too specific, we are loosing the power of this new system.

Apples to both versions